src/pyJiraCli/cmd_search.py

```diff
@@ -142,73 +164,94 @@ def _cmd_search(filter_str: str, results: int, save_file: str, server: Server) -
         LOG.print_info('Search string:', filter_str)
         LOG.print_info('Found Issues:', str(len(found_issues)))
 
-        if save_file is not None:
-
-            search_dict = {
-                'search': filter_str,
-                'max': results,
-                'found': len(found_issues)
-            }
+        search_dict = {
+            'search': filter_str,
+            'max': results,
+            'found': len(found_issues)
+        }
 
-            issue_list = []
-            jira = server.get_handle()
+        issue_list = []
+        jira = server.get_handle()
 
-            for issue in found_issues:
-                issue_dict = issue.raw
+        for issue in found_issues:
+            issue_dict = issue.raw
 
-                # Worklogs are requested for the issue.
-                if "worklog" in issue_dict["fields"]:
+            # Worklogs are requested for the issue.
+            if "worklog" in issue_dict["fields"]:
 
-                    # Get the worklogs for the issue
-                    # Itereate over all worklogs and store them in a list
-                    worklog_list = [
-                        log.raw for log in jira.worklogs(issue.key)]
+                # Get the worklogs for the issue
+                # Itereate over all worklogs and store them in a list
+                worklog_list = [
+                    log.raw for log in jira.worklogs(issue.key)]
 
-                    issue_dict["fields"]["worklog"] = {
-                        "total": len(worklog_list),
-                        "worklogs": worklog_list
-                    }
+                issue_dict["fields"]["worklog"] = {
+                    "total": len(worklog_list),
+                    "worklogs": worklog_list
+                }
 
-                issue_list.append(issue_dict)
+            issue_list.append(issue_dict)
 
-            search_dict['issues'] = issue_list
+        search_dict['issues'] = issue_list
 
+        if save_file is not None:
             ret_status = _save_search(save_file, search_dict)
-
         else:
-            _print_table(found_issues)
+            _print_table(search_dict, fields)
+            ret_status = Ret.CODE.RET_OK
 
     return ret_status
 
 
-def _print_table(issues: list) -> None:
-    """ Print a quick overview for all issues in the list.
+def _print_table(search_dict: dict, fields: list[str]) -> None:
+    """ Print a quick overview for all issues in the dict.
 
     Args:
-        issues (list): list with all found issues
+        search_dict (dict): dict with all found issues and search metadata.
     """
-    for header in HEADER:
-        print(f"{header:<{HEADER_COL_WIDTH[header]}}", end="")
+
+    if fields == []:
+        fields = DEFAULT_FIELDS
+
+    issues = search_dict['issues']
+    number_of_fields = min(len(fields), MAX_FIELDS_PRINTED)
+
+    print(f"{'Key':<{COLUMN_WIDTH}}", end="")
+
+    for field in range(number_of_fields):
```

Comment:
wouldn't it be something like:
`for idx in range(number_of_fields): ...`

Reply:
field_idx kombiniert eure Ansätze.